server/src/controllers/appointments.controller.js

```diff
@@ -34,7 +28,19 @@ export async function newAppointment(req, res, next) {
     }
 
     const { earlyTimeHour, lateTimeHour, address, email, ...rest } = req.body;
-    const coords = await geocodeAddress(address);
+    const geocodeData = await geocodeAddress(address);
+
+    if (geocodeData.status === "ZERO_RESULTS") {
+      res.status(404);
+      return next({ message: "The address submitted is not valid." });
+    }
+
+    if (!geocodeData.results.length) {
```

Comment:
```suggestion
    if (!geocodeData.results) {
```
FYI this works I think

Reply:
`results` will be an empty array so it won't be falsy